### ISSUE TITLE
Support array in struct?

### ISSUE THREAD

**Issue by XiangpengHao** [dtolnay/cxx#458]
What's your opinion on supporting array in struct? @dtolnay 
I assume this should be straightforward:

```rust
#[cxx::bridge]
mod ffi {
    struct Foo {
        a: [i32; 16],
        b: [Vec<usize>; 32],
    }
}
```

And it should be translated to 
```cpp
...
struct Foo;
struct Foo final {
  int32_t a[16];
  ::rust::Vec<size_t> b[32];
};
```

I'm not sure this is 100% safe as I'm new to both languages. If you're open to support this, I'll send a draft pull request.

**Comment by dtolnay** [comment]
I am open to supporting this.

`std::array` would probably be a friendlier equivalence.

```cpp
struct Foo final {
  std::array<int32_t, 16> a;
  std::array<rust::Vec<size_t>, 32> b;
};
```

**Comment by francesco-cattoglio** [comment]
Support for std::array would be awesome, I think it is one of the most useful stl bits imo.

**Comment by dtolnay** [comment]
Implemented in #503, released in cxx 1.0.4!
